var/spack/repos/builtin/packages/graphviz/package.py

```diff
@@ -93,31 +103,33 @@ def configure_args(self):
         spec = self.spec
         options = []
 
-        # These language bindings have been tested, we know they work.
-        tested_bindings = ('+java', '+perl')
+        need_swig = False
 
-        # These language bindings have not yet been tested.  They
-        # likely need additional dependencies to get working.
-        untested_bindings = (
-            '+swig', '+sharp', '+go', '+guile', '+io',
-            '+lua', '+ocaml', '+php',
-            '+python', '+r', '+ruby', '+tcl')
-
-        for var in untested_bindings:
+        for var in self.untested_bindings:
             if var in spec:
-                raise SpackException(
+                raise InstallError(
                     "The variant {0} for language bindings has not been "
                     "tested.  It might or might not work.  To try it "
                     "out, run `spack edit graphviz`, and then move '{0}' "
                     "from the `untested_bindings` list to the "
                     "`tested_bindings` list.  Be prepared to add "
                     "required dependencies.  "
                     "Please then submit a pull request to "
-                    "http://github.com/llnl/spack")
+                    "http://github.com/llnl/spack".format(var))
+            options.append('--disable-%s' % var[1:])
+
+        for var in self.tested_bindings:
+            if var in spec:
+                need_swig = True
+                enable = 'yes'
+            else:
+                enable = 'no'
+            options.append('--enable-%s=%s' % (var[1:], enable))
```

Comment:
Funny, I was chatting with a friend over beers last night about how, as I do more and more Python stuff, I'm loosing patience with the folks who trot out that saying....

And then there's `''.format()` vs `'' % (...)`....

Testing that change right now....

In this case I think that autoconf's command line arg flexibility is the root cause of the mess, it would all be cleaner if it only supported one way of doing it....